contracts/src/v0.8/feeds/DualAggregator.sol

```diff
@@ -512,24 +537,41 @@ contract DualAggregator is OCR2Abstract, OwnerIsCreator, AggregatorV2V3Interface
     emit CutoffTimeSet(s_cutoffTime);
   }
 
+  /**
+   * @notice sets the max sync iterations
+   * @param _maxSyncIterations new max sync iterations
+   */
+  function setMaxSyncIterations(
+    uint32 _maxSyncIterations
+  ) external onlyOwner {
+    s_maxSyncIterations = _maxSyncIterations;
+    emit MaxSyncIterationsSet(s_maxSyncIterations);
+  }
+
   /**
    * @notice sync data with the primary rounds, return the freshest valid round id
    */
   function _getSyncPrimaryRound() internal view returns (uint32 roundId) {
-    // get the latest round id
+    // get the latest round id and the max iterations
     uint32 latestAggregatorRoundId = s_hotVars.latestAggregatorRoundId;
+    uint32 maxIterations = s_maxSyncIterations;
 
     // decreasing loop from the latest primary round id
     for (uint32 round_ = latestAggregatorRoundId; round_ > 0; --round_) {
-      Transmission memory transmission = s_transmissions[round_];
+      // in case the loop reached the maxIterations, break it
+      if (latestAggregatorRoundId - round_ == maxIterations) {
+        break;
+      }
 
-      // check if this round does not accomplish the cutoff time condition
-      if (transmission.recordedTimestamp + s_cutoffTime < block.timestamp) {
+      // in case it's the latest secondary round id, return it
+      if (round_ == s_hotVars.latestSecondaryRoundId || latestAggregatorRoundId - round_ == maxIterations) {
```

Comment:
Isn't the second condition excluded by the previous if?

Reply:
You're correct, I fixed it in [this PR](https://github.com/smartcontractkit/chainlink/pull/15243).